### ISSUE TITLE
[Feature Request] Make "forward" handle large inputs in batches

### ISSUE THREAD

**Issue by MaigoAkisame** [pytorch/pytorch#3580]
Keras has this nice "[model.predict](https://keras.io/models/model/#predict)" function which runs the forward pass on the input in batches. So I can feed a large data tensor (e.g. the entire test set) into a model and get the predictions in one function call.

It would be nice if PyTorch models could do the same. For example, if a model has only one input and one output, the "predict" function could be implemented as follows:

    def predict(self, x, batch_size = 32):
        output = []
        for i in range(0, len(x), batch_size):
            output.append(self.forward(x[i : i + batch_size]))
        return torch.cat(output)

The code will be a little more complex if there are multiple inputs or outputs.

**Comment by fmassa** [comment]
I'm not sure we want to add such convenience function (but others might have different opinions).
I think that there can be weird edge cases (for example, the batch might not be the first dimension as in some rnns I think), and for the user it can be implemented in one line
```python
torch.cat([model(x) for x in input.split(batch_size, 0)], 0)
```
What do you think?

**Comment by MaigoAkisame** [comment]
That's a nice one-liner for the case of a single input and a single output. Thanks!

Let's see what others think.

**Comment by apaszke** [comment]
As @fmassa has shown, this is really easy to implement in a single line of code, and this makes it a lot clearer and more explicit. I don’t think we should have such helpers
